src/MyRoad.Domain/EmployeesLogs/EmployeeLogService.cs

```diff
@@ -0,0 +1,171 @@
+        {
+            var result = await _employeeLogValidator.ValidateAsync(employeelog);
+            if (!result.IsValid)
+            {
+                return result.ExtractErrors();
+            }
+
+            var employee = await employeeRepository.GetByIdAsync(employeelog.EmployeeId);
+            if (employee == null)
+            {
+                return EmployeeErrors.NotFound;
+            }
+
+            var user = await userRepository.GetByIdAsync(employeelog.CreatedByUserId);
+            if (user is null)
+            {
+                return UserErrors.NotFound;
+            }
+
+
+            await unitOfWork.BeginTransactionAsync();
+
+            try
+            {
+                switch (user.Role)
+                {
+                    case UserRole.Admin when userContext.Role == UserRole.Admin:
+                        employee.TotalDueAmount += employeelog.DailyWage;
+                        break;
+                    case UserRole.Manager when userContext.Role == UserRole.Manager:
+                        break;
+                    default:
+                        return UserErrors.UnauthorizedUser;
+                }
+
+                await employeeLogRepository.CreateAsync(employeelog);
+                await employeeRepository.UpdateAsync(employee);
+                await unitOfWork.CommitTransactionAsync();
+
+                return new Success();
+            }
+            catch (Exception)
+            {
+                await unitOfWork.RollbackTransactionAsync();
+                throw;
+            }
+        }
+
+        public async Task<ErrorOr<Success>> UpdateAsync(EmployeeLog employeelog)
+        {
+            var validationResult = await _employeeLogValidator.ValidateAsync(employeelog);
+            if (!validationResult.IsValid)
+            {
+                return validationResult.ExtractErrors();
+            }
+
+            var existingEmployeeLog = await employeeLogRepository.GetByIdAsync(employeelog.Id);
+            if (existingEmployeeLog is null || existingEmployeeLog.IsDeleted)
+            {
+                return EmployeeLogErrors.NotFound;
+            }
+
+            var employee = await employeeRepository.GetByIdAsync(employeelog.EmployeeId);
+            if (employee is null|| !employee.Status)
+            {
+                return EmployeeErrors.NotFound;
+            }
+
+            var prevDailyWage = existingEmployeeLog.DailyWage;
+            var newDailyWage = employeelog.DailyWage;
+
+
+            try
+            {
+                await unitOfWork.BeginTransactionAsync();
+                employee.TotalDueAmount = employee.TotalDueAmount - prevDailyWage + newDailyWage;
+                existingEmployeeLog.MapUpdateEmployeeLog(employeelog);
+
+                await employeeRepository.UpdateAsync(employee);
+                await employeeLogRepository.UpdateAsync(existingEmployeeLog);
+                await unitOfWork.CommitTransactionAsync();
+
```

Comment:
Before you make an update, you must ensure that this update does not cause problems in the employee’s financial accounts.
example : 

```suggestion
               var newTotalDueAmount = employee.TotalDueAmount - prevDailyWage + newDailyWage;
                if (employee.TotalPaidAmount > newTotalDueAmount)
                {
                    return Error;
                }
```